src/librustdoc/html/static/css/rustdoc.css

```diff
@@ -891,8 +891,10 @@ so that we can apply CSS-filters to change the arrow color in themes */
 	color: var(--search-results-grey-color);
 }
 .search-results .result-name .typename {
+	display: inline-block;
 	color: var(--search-results-grey-color);
 	font-size: 0.875rem;
+	width: 6rem;
```

Comment:
I picked this number arbitrarily. Two issues:

1. It would be nice if it weren't based on eyeballing the layout.

2. This isn't actually wide enough for "associated constant". If I increase this to 9em, it fits, but then all the other rows have excessive spacing. I see two solutions: one is to shorten the typename to just "constant", the other is to make the spacing somehow adaptive to what the longest typename is. But I'm not sure how to implement that.

Reply:
What happens when it overflows? Does it wrap and just become a taller row? I think that honestly wouldn't be terrible if that's what happens.

Reply:
It's not terrible, but still looks bad IMO:

![image](https://github.com/rust-lang/rust/assets/37223377/de69e7a9-934d-4dc5-821a-1f49ee05a316)

Reply:
:/ true -- i wonder if vertically center-aligning those rows would help.

Reply:
I think we really need to just keep rows a constant height. Variable height rows almost always look a bit off.

Reply:
Something that could at least make them look better is to make sure both the path/name and short description are aligned the same, to the top or the middle as Michael suggested.

Reply:
I was able to avoid the wrapping by adding a tiny bit extra space.